docs/Optimization.md

```diff
@@ -1,52 +1,63 @@
 # Optimization
 
-## Memoizing Middleware
+## Memoization
 
-One optimization that flipper provides is a memoizing middleware. The memoizing middleware ensures that you only make one adapter call per feature per request. This means if you check the same feature over and over, it will only make one Mongo, Redis, or whatever call per feature for the length of the request.
+By default, Flipper will preload and memoize all features to ensure one adapter call per request. This means no matter how many times you check features, Flipper will only make one network request to Mongo, Redis, or whatever adapter you are using for the length of the request.
 
-You can use the middleware like so for Rails:
+### Preloading
+
+By default, Flipper will preload all features before each request. This is recommended if you have a limited number of features (< 100?) and they are used on most requests. If you have a lot of features, but only a few are used on most requests, you may want to customize preloading:
 
 ```ruby
-# setup default instance (perhaps in config/initializers/flipper.rb)
+# config/initializers/flipper.rb
 Flipper.configure do |config|
-  config.default do
-    Flipper.new(...)
-  end
+  # Load specific features that are used on most requests
+  config.preload = [:stats, :search, :some_feature]
+
+  # Disable preloading
+  config.preload = false
 end
+```
+
+Features that are not preloaded are still memoized, ensuring one adapter call per feature during a request.
+
+### Skip memoization
 
-# This assumes you setup a default flipper instance using configure.
-Rails.configuration.middleware.use Flipper::Middleware::Memoizer
+Prevent preloading and memoization by setting `unless` to a proc that evaluates to true.
+
+```ruby
+# config/initializers/flipper.rb
+Flipper.configure do |config|
+  config.memoize_unless = ->(request) { request.path.start_with?("/assets") }
```

Comment:
Would it make more sense to just sent `config.memoize` to a block, and the result 


```ruby
Flipper.configure do |config|
  # Memoize all requests (default)
  config.memoize = true 

  # Disable memoization
  config.memoize = false 

  # Memoize if not an asset request
  config.memoize = ->(request) { !request.path.start_with?("/assets") }
end
```

This feels cleaner to me than having both `memoize` (boolean) and `memoize_unless` (block)